src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineComputerLauncher.java

```diff
@@ -214,19 +218,23 @@ public void launch(SlaveComputer slaveComputer, TaskListener listener) {
             launch(computer, listener);
         } catch (IOException ioe) {
             ioe.printStackTrace(listener.error(ioe.getMessage()));
-            node = (ComputeEngineInstance) slaveComputer.getNode();
-            if (node != null) {
-                try {
-                    node.terminate();
-                } catch (Exception e) {
-                    listener.error(String.format("Failed to terminate node %s", node.getDisplayName()));
-                }
-            }
+            terminateNode(slaveComputer, listener);
         } catch (InterruptedException ie) {
 
         }
     }
 
+    private static void terminateNode(SlaveComputer slaveComputer, TaskListener listener) {
+        ComputeEngineInstance node = (ComputeEngineInstance) slaveComputer.getNode();
+        if (node != null) {
```

Comment:
Should we add some fine logs if `node == null`, something like `"Tried to terminate unknown node from computer %s"` ?

Reply:
Mmmmh yes, we could.

Reply:
done into ce2791c144537bd34b83b04c159b5e7cb471c9f3